### DISCUSSION TITLE
Disappointed: My *arrs sit behind a reverse-proxy. I don't require an API key to access them

### DISCUSSION THREAD

**Discussion by serendrewpity** [trossr32/sonarr-radarr-lidarr-autosearch-browser-extension#215] [Ideas]
Disappointed: My *arrs sit behind a reverse-proxy. I don't require an API key to access them. So, I can simply navigate to hxxp://www.example.com/*arr and get to radarr, sonarr, and lidarr. But I can't use your extension because it requires me to enter the API key. Which I can supply, but I'd rather not.

**Comment by trossr32** [comment]
You don't need to provide an API key for the extension to work. The only reason the extension uses an API key at the moment is to identify the version of *arr that you're using so that it can set the correct _Search Path URL_ and _Search Field Selector_ in the settings in the _Advanced_ settings tab.

These settings should default to the correct values for the latest versions of *arr, so as long as your *arr instances are up-to-date you should be good to go.

Obviously you'll still need to set the _Protocol, domain and port_ so that the extension can redirect to your *arr instances from integration sites like IMDb.

Here's where the *arr API is called, as you can see it's just getting version info: https://github.com/trossr32/sonarr-radarr-lidarr-autosearch-browser-extension/blob/master/src/content/js/core.js#L747

![image](https://github.com/user-attachments/assets/1e844bd5-437c-4e3c-a548-204c1e2a8eef)
